spec/test_data/v5.1/expected_active_record_relation.rbi

```diff
@@ -162,4 +162,10 @@ class ActiveRecord::Associations::CollectionProxy < ActiveRecord::Relation
 
   sig { params(attributes: T.untyped, block: T.nilable(T.proc.params(object: Elem).void)).returns(Elem) }
   def create!(attributes = nil, &block); end
+
+  sig { params(args: T.untyped).returns(Elem) }
```

Comment:
Interesting that these methods are only needed for `ActiveRecord::Associations::CollectionProxy`. Is it working for `ActiveRecord::Relation` and `ActiveRecord::AssociationRelation` because they use the signatures from L34 to L59?

Reply:
I left a more detailed comment but for this specifically:
- we're defining the methods directly on `ActiveRecord::Relation` so sorbet will use that
- `ActiveRecord::AssociationRelation` doesn't redefine `find` so when sorbet auto-generates the RBI is won't add that method to `ActiveRecord::AssociationRelation` and thus it will fallback to the inherited method and sig on `ActiveRecord::Relation`
- `ActiveRecord::Associations::CollectionProxy` actually redefines `find` (I think it just winds up calling super but might do something slightly different) which cases sorbet to generate that method when auto-generating the RBI. This would cause sorbet to use _that_ unsigged method definition instead of the inherited one so we need to explicitly define it ourselves here.